arangod/Graph/Providers/SingleServerProvider.h

```diff
@@ -147,6 +171,9 @@ class SingleServerProvider {
   SingleServerBaseProviderOptions _opts;
 
   RefactoredTraverserCache _cache;
+  bool _useVertexCache = true;
+  FoundVertexCache _vertexCache;
```

Comment:
Can we make this an optional and therefore get rid of the additional boolean variable?

Reply:
Then the code needs substantial rewriting. The trouble is that I use an iterator pointing to `_vertexCache.end()` in two cases: 1) the cache is used but no entry is found, 2) the cache is not used at all. This is because exactly the same has to happen then. And only after we got the data from the iterator do we need to distinguish the cases again. I fear that it leads to quite a bit of code duplication if we rewrite this. So I would prefer to keep it this way.

Reply:
Done with the move to `shared_ptr`s below.